analysis/analyse.go

```diff
@@ -779,25 +788,25 @@ func findErrorCodesFromIdentTaint(c *context, visitedIdents map[*ast.Object]stru
 	}
 
 	for _, destruct := range taintResult.destructAssignment {
-		callExpr, ok := astutil.Unparen(destruct.source).(*ast.CallExpr)
-		if !ok {
-			panic("should be unreachable: destructirung assignment should only originate from a call expression.")
-		}
+		switch expr := astutil.Unparen(destruct.source).(type) {
+		case *ast.TypeAssertExpr:
+			newCodes := findErrorCodesInExpression(c, visitedIdents, expr.X, function)
+			result = Union(result, newCodes)
+		case *ast.CallExpr:
+			funType := pass.TypesInfo.TypeOf(expr.Fun).(*types.Signature) // function of call expression should always be of type signature.
 
-		funType, ok := pass.TypesInfo.TypeOf(callExpr.Fun).(*types.Signature)
-		if !ok {
-			panic("should be unreachable: function of call expression should always be of type signature.")
-		}
+			// Destructuring mode.
+			// We're going to make some crass simplifications here, and say... if this is anything other than the last arg, you're not supported.
+			if destruct.position != funType.Results().Len()-1 {
+				pass.ReportRangef(destruct.target, "unsupported: tracking error codes for function call with error as non-last return argument")
+				continue
+			}
 
-		// Destructuring mode.
-		// We're going to make some crass simplifications here, and say... if this is anything other than the last arg, you're not supported.
-		if destruct.position != funType.Results().Len()-1 {
-			pass.ReportRangef(destruct.target, "unsupported: tracking error codes for function call with error as non-last return argument")
-			continue
+			newCodes := findErrorCodesInCallExpression(c, expr, function)
+			result = Union(result, newCodes)
+		default:
+			panic(fmt.Sprintf("destruct assignment should only originate from a call expression: type %T", expr))
```

Comment:
```suggestion
			panic(fmt.Sprintf("destructuring assignment is currently only supported from a call expression: found one in an expression of type %T", expr))
```

I started with a typo fix...

... but also suggest a different wording, because I'm pretty sure there's more origins of these, fwiw.  `x, y := w, z` expressions exist and I think they're filed under destructuring assignments.  (Pushing that further support doesn't necessarily have to be part of this PR though.)

Reply:
That seems fair. I'll open an issue for it so we can track that.